### ISSUE TITLE
additional enhancements to Bootstrap project

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#596]
Rather than list these as separate issues I think the following list of enhancements to the Bootstrap project template would be useful:

- [x] Add an optional script property to the LO icon like with XOT projects. In the site template we have styles at that level but not script.
- [x] Add a styles option under advanced options. I know we have New HTML Code but you have to add `<styles></styles>` to that rather than just the css.
- [x] Add option to determine where page navbar sits e.g. at top as is now, below jumbotron container, or hidden
- [x] Add an optional footer property e.g. that sits above current footer and can be populated with anything in the editor.
- [x] Add an option to hide the social networking links. Perhaps even control of individual icons.
- [x] Add option to hide jumbotron

**Comment by ronm123** [comment]
@JohnSmith-LT @FayCross I think this list of enhancements is nearly complete now. Just two or three issues in what I added today:

1. Would be good to have the ability to disable individual properties in a group as discussed with John via #661

2. I also added a comment to #661 about the navbar text colour and hover colour options I've added but where the hover colour isn't changing for some reason.

3. I've also added footer options which work and display whatever styling is added to the wysiwyg editor but not images. For instant I think users might quite like the option to add a panel of logos via the editor and these do show in the editor window but I'm not sure what I need to get those to display in the footer itself? Does the FileLocation value need to be converted somehow?

**Comment by JohnSmith-LT** [comment]
Hi @ronm123, I've committed some code that fixes 2 and 3... i think!! You can let me know...

As for 1, that's going to be a bit more involved and I'm not sure when i'd have the time to look at that, at least this week. Also, i'm wanting to hold off to hear what @FayCross thinks about it since she coded the group functionality and might have some better ideas as to how we could achieve that...

**Comment by ronm123** [comment]
Thanks @JohnSmith-LT that's great. Both changes work well although I changed the order of the convert image paths for the custom footer so that it works regardless of the position option selected.

As for issue 1 I've changed the default values of each of the colour pickers to 0xEEEEEE to match the background colour of the wizard so almost looks disabled and then changed the conditions to ignore those values if they haven't been changed. That does mean that 0xEEEEEE can't ever be used via the pickers but I think that's better than having the default picker values override the default theme colours when the groups are added. As discussed someone might add the header group to add logos but not want to change the header colours. The only other solution I thought of was to change the fields to text rather than pickers but I think this is better all round. Hopefully you and @FayCross agree this is ok and it's certainly a simpler option than adding enable/disable functionality for each property?

I think all these enhancements are done now so will close the issue.

**Comment by ronm123** [comment]
BTW I added an option to change the footer background colour and although I included the check for two values I didn't add all the code to apply gradients. Probably not needed for footer background and if we thought it was probably better to make all that a single function rather than repeat for each property group?

**Comment by FayCross** [comment]
Sorry I committed a different fix for the colour picker issues without reading all of this. It doesn't allow you to properly turn off options like @JohnSmith-LT suggests - perhaps this is something we could add when we have more time?

@ronm123 would you mind if I renamed the navbar options? I'm not sure that's the clearest name we could give it for most users. Perhaps 'page menu' or 'navigation bar' would be better?

**Comment by JohnSmith-LT** [comment]
That's fine @FayCross, haven't had a chance to try your fix yet but I knew what I had suggested was a bit more involved so yes, perhaps as an enhancement further down the line. My only thought would be, if you add colours to try and then decide you don't want them can you clear the field again?

Also, this might be a good place to use the new group information box to leave some instructions?

**Comment by ronm123** [comment]
Hi @FayCross @JohnSmith-LT sorry have been busy with online meetings so far today and now another priority for the next hour so no time to test yet but will do later.

@FayCross sure renaming what the authors see is fine but perhaps leave what the code picks up as navbar? I didn't wonder about various names but Page menu could easily be confused with the sections menu so I think the word bar is important to include. Navigation bar sounds fine to me.

**Comment by FayCross** [comment]
Yes you can clear the colour again after you've selected one @JohnSmith-LT 
& no, I'll just change the wording on screen Ron so nothing in the code

**Comment by ronm123** [comment]
Hi @FayCross I've pulled down the changes and am testing as we speak.
Empty colour pickers seem to work great. For future reference how/where did you enable that?

The change of label to Navigation bar seems fine and the ability to hide it and change background, text and hover colour still works but the navbar position no longer works and I can't see why. e.g. regardless of setting my navbar is showing below the header rather than the default of above the header. I'll explore further but any ideas?

**Comment by FayCross** [comment]
Strange... the position option still works for me. I'm pretty sure the only change I made was to the name that appears in the editor. I'll check though.

I added an option to the code that calls the jscolor plugin so that it allows blank values and then removed the default values from the xwd. Previously if the colour picker received no value then it would show as black #000000. So if there are any other places where you'd like the colour picker to be blank then you can just delete the default value from the xwd.

**Comment by ronm123** [comment]
@FayCross If I create a new project the navbar position works fine. But with the projects I was using to add all this I can't get it to work even if I remove the navbar group and then re-add. If I do that I can see the xml change but not the playback. I've force refreshed etc but will explore further.

Curiously if I put an alert in application.js do show me the value it's picking up the alert flashes up then dissapears whereas it should stay on screen until click ok which again had been working previously.

**Comment by ronm123** [comment]
@FayCross found it and my silly mistake! When I was testing the theme I added and then added the ability to have a theme.js file too I had the jquery to move the navbar in that js file for testing the theme js was working. I may have even committed the theme with that jquery in the js file so I'll check and remove if I did.

So sorry this isn't an issue it's working fine just not with the black grey theme selected until I remove that jquery.
